lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -1628,31 +1643,50 @@ struct ObjectSubfieldOpConversion
   const DenseMap<StringAttr, firrtl::ClassType> &classTypeTable;
 };
 
-struct ClassFieldOpConversion : public OpConversionPattern<ClassFieldOp> {
+LogicalResult convertClassFieldsTypes(ClassFieldsLike oldOp,
+                                      ClassFieldsLike newOp,
+                                      const TypeConverter *typeConverter) {
+  llvm::SmallVector<NamedAttribute> fieldTypes;
+  llvm::SmallVector<StringAttr> fieldNames = oldOp.getFieldNames();
+  mlir::DictionaryAttr fieldIdxs =
+      mlir::cast<mlir::DictionaryAttr>(oldOp->getAttr("fieldIdxs"));
+  for (auto field : fieldNames) {
+    std::optional<Type> type = oldOp.getFieldType(field);
+    if (!type.has_value())
+      return failure();
+    fieldTypes.push_back(mlir::NamedAttribute(
+        cast<StringAttr>(field),
+        mlir::TypeAttr::get(typeConverter->convertType(type.value()))));
+  }
+  auto *ctx = newOp.getContext();
+  newOp->setAttr("fieldNames", mlir::ArrayAttr::get(ctx, {fieldNames.begin(),
+                                                          fieldNames.end()}));
+  newOp->setAttr("fieldTypes", mlir::DictionaryAttr::get(ctx, fieldTypes));
+  newOp->setAttr("fieldIdxs", fieldIdxs);
+  return llvm::success();
```

Comment:
super nit: just like you can use `failure()` above, you could use `success()` without the `llvm::` namespace.